pkg/kubectl/cmd/get_test.go

```diff
@@ -358,6 +359,66 @@ func TestGetSortedObjects(t *testing.T) {
 
 }
 
+func TestGetSortedReplicaSets(t *testing.T) {
+	rss := &extensions.ReplicaSetList{
+		ListMeta: unversioned.ListMeta{
+			ResourceVersion: "15",
+		},
+		Items: []extensions.ReplicaSet{
+			{
+				ObjectMeta: api.ObjectMeta{Name: "s", Namespace: "test", ResourceVersion: "10"},
+				Spec:       extensions.ReplicaSetSpec{},
+			},
+			{
+				ObjectMeta: api.ObjectMeta{Name: "r", Namespace: "test", ResourceVersion: "11"},
+				Spec:       extensions.ReplicaSetSpec{},
+			},
+			{
+				ObjectMeta: api.ObjectMeta{Name: "t", Namespace: "test", ResourceVersion: "9"},
+				Spec:       extensions.ReplicaSetSpec{},
+			},
+		},
+	}
+
+	f, tf, codec, ns := NewAPIFactory()
+	codec = testapi.Extensions.Codec()
+	tf.Printer = &testPrinter{}
+	tf.Client = &fake.RESTClient{
+		NegotiatedSerializer: ns,
+		Resp:                 &http.Response{StatusCode: 200, Header: defaultHeader(), Body: objBody(codec, rss)},
+	}
+	tf.Namespace = "test"
+	tf.ClientConfig = &restclient.Config{ContentConfig: restclient.ContentConfig{GroupVersion: &unversioned.GroupVersion{Group: "extensions", Version: "v1beta1"}}}
+
+	buf := bytes.NewBuffer([]byte{})
+	errBuf := bytes.NewBuffer([]byte{})
+
+	cmd := NewCmdGet(f, buf, errBuf)
+	cmd.SetOutput(buf)
+
+	// sorting with metadata.name
+	cmd.Flags().Set("sort-by", ".metadata.name")
```

Comment:
You should also test `--sort-by=.metadata.creationTimestamp` that causes the panic in the issue.